loadtest/control/actions.go

```diff
@@ -486,3 +486,145 @@ func ViewUser(u user.User) UserActionResponse {
 	}
 	return UserActionResponse{Info: fmt.Sprintf("viewed user %s", member.UserId)}
 }
+
+// Reload simulates the given user reloading the page
+// while connected to the server by executing the API
+// calls seen during a real page reload.
+func Reload(u user.User) UserActionResponse {
+	// Getting preferences.
+	err := u.GetPreferences()
+	if err != nil {
+		return UserActionResponse{Err: NewUserError(err)}
+	}
+
+	prefs, _ := u.Store().Preferences()
+	var userIds []string
+	chanId := ""
+	for _, p := range prefs {
+		switch {
+		case p.Name == model.PREFERENCE_NAME_LAST_CHANNEL:
+			chanId = p.Value
+		case p.Category == model.PREFERENCE_CATEGORY_DIRECT_CHANNEL_SHOW:
+			userIds = append(userIds, p.Name)
+		}
+	}
+
+	if chanId != "" {
+		// Marking the channel as viewed
+		_, err := u.ViewChannel(&model.ChannelView{
+			ChannelId:     chanId,
+			PrevChannelId: "",
+		})
+		if err != nil {
+			return UserActionResponse{Err: NewUserError(err)}
+		}
+	}
+
+	if ok, err := u.IsSysAdmin(); ok && err != nil {
+		err = u.GetConfig()
```

Comment:
It looks like we are discarding error of `u.IsSysAdmin()`. Also it sounds a little bit odd to have both `ok` and `err` at the same time.

Reply:
The error is needed cause to figure if the user is admin we ask the store and that could potentially fail for other reasons. The `ok` value would be instead the `bool` telling us if the user is admin. So we want to run the block only if we didn't get any error and `ok == true`. 
We could indeed log in case of error, we just didn't do it because our current implementation of the store (`memstore`) doesn't really allow for errors.

Reply:
I see, but it is error-prone. What if we change the store logic?

Reply:
I think we just need to fix this. The check should be `if ok, err := u.IsSysAdmin(); ok && err == nil`.